src/v/pandaproxy/server.h

```diff
@@ -45,6 +46,8 @@ class server {
         ssx::semaphore& mem_sem;
         ss::abort_source as;
         ss::smp_service_group smp_sg;
+        credential_t user;
+        config::rest_authn_type authn_type;
```

Comment:
This is the server context.

* The `request_t` is probably a better place for `user`, otherwise it will keep getting overwritten.. Perhaps `auth_result` is a better type to attach to the request.
* `authn_type` is per-listener, this will be overwritten too.

Reply:
>     * The `request_t` is probably a better place for `user`, otherwise it will keep getting overwritten.. Perhaps `auth_result` is a better type to attach to the request.

Tried using the auth_result but that object throws in the dtor which leads to compile-time errors at the seastar level for `sharded<T>.invoke_on()`. See [slack thread](https://redpandadata.slack.com/archives/C01ND4SVB6Z/p1663790753852919) for details.